### ISSUE TITLE
predict does not work with tfjs-node but with tfjs-node-gpu

### ISSUE THREAD

**Issue by timoklingenhoefer** [tensorflow/tfjs#2754]
#### TensorFlow.js version
"@tensorflow/tfjs-node": "^1.5.2",
"@tensorflow/tfjs-node-gpu": "^1.5.2",

#### Describe the problem or feature request
I have a model that runs on tfjs-node-gpu, but then I changed to using tfjs-node to test on a different machine without CUDA, it fails to run with this error:
```
Error: Invalid TF_Status: 3
Message: indices[0] = 6 is not in [0, 6)
    at NodeJSKernelBackend.executeSingleOutput (C:\redacted\project-name\node_modules\@tensorflow\tfjs-node\dist\nodejs_kernel_backend.js:193:43)
    at NodeJSKernelBackend.gather (C:\redacted\project-name\node_modules\@tensorflow\tfjs-node\dist\nodejs_kernel_backend.js:1172:21)
    at engine_1.ENGINE.runKernelFunc.x (C:\redacted\project-name\node_modules\@tensorflow\tfjs-core\dist\ops\segment_ops.js:115:27)
    at C:\redacted\project-name\node_modules\@tensorflow\tfjs-core\dist\engine.js:528:55
    at C:\redacted\project-name\node_modules\@tensorflow\tfjs-core\dist\engine.js:388:22
    at Engine.scopedRun (C:\redacted\project-name\node_modules\@tensorflow\tfjs-core\dist\engine.js:398:23)
    at Engine.tidy (C:\redacted\project-name\node_modules\@tensorflow\tfjs-core\dist\engine.js:387:21)
    at kernelFunc (C:\redacted\project-name\node_modules\@tensorflow\tfjs-core\dist\engine.js:528:29)
    at C:\redacted\project-name\node_modules\@tensorflow\tfjs-core\dist\engine.js:539:27
    at Engine.scopedRun (C:\redacted\project-name\node_modules\@tensorflow\tfjs-core\dist\engine.js:398:23)

```

#### Code to reproduce the bug / link to feature request
only change from
```
import * as tf from '@tensorflow/tfjs-node-gpu';
```
to
```
import * as tf from '@tensorflow/tfjs-node';
```

I found that this can be caused by wrong input data https://github.com/tensorflow/tensorflow/issues/15091#issuecomment-435140534
but this souldn´t be the case as I am using ```tf.node.decodeImage(...)``` to get my input tensor

**Comment by rthadur** [comment]
@timoklingenhoefer  thank you , can you please provide these details 
                    1) Model if possible 
                    2) Operating system

**Comment by timoklingenhoefer** [comment]
Im am running win 7 
with this model I downloaded from [here](https://raw.githubusercontent.com/shaqian/tfjs-yolo-demo/master/dist/model/v3/model.json)

**Comment by timoklingenhoefer** [comment]
ok I solved it myself
As I tried to make an minimal version of my application for replication, I was unable to produce the error. 
It turns out that the prdiction was working fine, but my post-processing of the output caused the issue. 
Basically my post-processor tried to gather data from a tensor that didn´t exist. 

```
outputTensor.gather(tf.tensor1d(mask[i], 'int32'));
```
outputTensor only had 6 entries and mask tried to access the indices 6, 7 and 8. 

So it was indeed caused by wrong data, but at a different place than expected.
